lp-tool/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	influxSer "github.com/influxdata/telegraf/plugins/serializers/influx"
+)
+
+
+// Custom type to handle multiple tag flags
+type TagMap map[string]string
+
+// Implement String for TagMap ; required by flag.Value interface
+func (tm TagMap) String() string {
+	var pairs []string
+	for k, v := range tm {
+		pairs = append(pairs, fmt.Sprintf("%s=%s", k, v))
+	}
+	return strings.Join(pairs, ", ")
+}
+
+// Implement Set for TagMap ; required by flag.Value interface
+func (tm TagMap) Set(value string) error {
+	parts := strings.SplitN(value, "=", 2)
+	if len(parts) != 2 {
+		return fmt.Errorf("tag must be in format 'key=value', got: %s", value)
+	}
+
+	key := strings.TrimSpace(parts[0])
+	val := strings.TrimSpace(parts[1])
+
+	if key == "" {
+		return fmt.Errorf("tag key cannot be empty")
+	}
+
+	if val == "" {
+		return fmt.Errorf("tag value cannot be empty")
+	}
+
+    if _, exists := tm[key]; exists {
+		return fmt.Errorf("tag key already provided")
+	}
+
+	tm[key] = val
+
+	return nil
+}
+
+type Config struct {
+	InputFile  string
+	OutputFile string
+    Tags TagMap
+}
+
+func main() {
+    config, err := parseFlags()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		flag.Usage()
+		os.Exit(1)
+	}
+	if err := processLineProtocol(config); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("Successfully processed %s and wrote to %s\n", config.InputFile, config.OutputFile)
+}
+
+func parseFlags() (Config, error) {
+	var config Config
+	config.Tags = make(TagMap)
+
+	flag.StringVar(&config.InputFile, "input", "holochain.influx", "Input line protocol file path")
+	flag.StringVar(&config.OutputFile, "output", "holochain.tmp.influx", "Output line protocol file path")
+	flag.Var(&config.Tags, "tag", "Add a tag in format 'key=value' (can be used multiple times)")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [options]\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "A CLI tool to process InfluxDB line protocol files for adding custom tags.\n\n")
+		fmt.Fprintf(os.Stderr, "Options:\n")
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "\nExample:\n")
+		fmt.Fprintf(os.Stderr, "  %s -input data.influx -output holochain.influx -tag env=production -tag region=europe -tag version=1.2.3\n", os.Args[0])
+	}
+
+	flag.Parse()
+
+	if config.InputFile == "" || len(config.Tags) == 0 {
+		return config, fmt.Errorf("input file or tags not specified")
+	}
+    if config.OutputFile == "" {
+        config.OutputFile = config.InputFile + ".tmp.influx"
+    }
```

Comment:
Shouldn't this also just be an error? It seems strange that someone could specify `-output=""` and they get a seemingly random output file.

```suggestion
    if config.OutputFile == "" {
		return config, fmt.Errorf("output file cannont be an empty path")
    }
```